subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -46,73 +47,113 @@
+        public String getDefaultVersionNotation() {
+            return getNotationForVersion(version);
+        }
+
+        public String getNotationForVersion(@Nullable String specified) {
+            return group + ":" + name + (specified != null ?  ":" + specified : "");
+        }
+    }
+
+    // Should be the same as the version listed in the junit-bom corresponding to the default junit-jupiter version.
+    // Only used here for testing, but defined in this class to keep it near the default junit-jupiter version.
+    @VisibleForTesting
+    public static final String JUNIT_PLATFORM_DEFAULT_VERSION = "1.8.2";
+
+    @VisibleForTesting
+    public enum TestingFramework {
+        // When updating these versions, be sure to update the default versions noted in `JvmTestSuite` javadoc
+        JUNIT4("junit", "junit", "4.13.2"),
+        JUNIT_JUPITER("org.junit.jupiter", "junit-jupiter", "5.8.2", Collections.singletonList(
+            // junit-jupiter's BOM, junit-bom, specifies the platform version
+            new Module("org.junit.platform", "junit-platform-launcher", null)
+        )),
+        SPOCK("org.spockframework", "spock-core", "2.1-groovy-3.0"),
+        KOTLIN_TEST("org.jetbrains.kotlin", "kotlin-test-junit5", "1.7.10", Collections.singletonList(
+            // kotlin-test-junit5 depends on junit-jupiter, which in turn specifies the platform version
+            new Module("org.junit.platform", "junit-platform-launcher", null)
+        )),
+        TESTNG("org.testng", "testng", "7.5");
+
+        private final Module module;
+        private final List<Module> dependencies;
+
+        TestingFramework(String group, String name, String defaultVersion) {
+            this(group, name, defaultVersion, Collections.emptyList());
         }
 
-        @Nullable
-        public String getDependency() {
-            return getDependency(getDefaultVersion());
+        TestingFramework(String group, String name, String defaultVersion, List<Module> dependencies) {
+            this.module = new Module(group, name, defaultVersion);
+            this.dependencies = dependencies;
         }
 
-        @Nullable
-        public String getDependency(String version) {
-            if (null != module) {
-                return module + ":" + version;
-            } else {
-                return null;
+        public String getDefaultVersion() {
+            return module.version;
+        }
+
+        public List<Dependency> getDependencies(String version, DependencyFactory dependencyFactory) {
+            List<Dependency> deps = new ArrayList<>(1 + dependencies.size());
+            deps.add(dependencyFactory.create(module.getNotationForVersion(version)));
+            for (Module dep : dependencies) {
+                // TODO: Should we separate these between implementation and runtime dependencies?
+                // Do we need junit-platform-launcher on the compile classpath or only during runtime?
```

Comment:
IMO its better to take care of this now so that in the future we don't have to worry about users depending on JUnit Platform stuff at compile-time. Updated to split the implementation and runtime dependencies.